lib/kitchen/driver/ec2.rb

```diff
@@ -134,15 +133,34 @@ def default_public_ip_association
 
       private
 
+      def aws_credentials
+        if config[:use_iam_profile]
+          {
+            :use_iam_profile => true
+          }
+        else
+          credentials = {}
+          [:aws_access_key_id, :aws_secret_access_key].each { |key| 
+            if !config.key?(key) or config[key].nil? or config[key].to_s.empty?
+              raise Kitchen::UserError, "Must provide #{key} or use IAM profile"
```

Comment:
I would remove the if/else block, leave the error as 
raise Kitchen::UserError, "Must provide #{key} or use IAM profile" unless config[key]